sszuint64.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	fmt "fmt"
+
+	fssz "github.com/ferranbt/fastssz"
+)
+
+var _ fssz.HashRoot = (SSZUint64)(0)
+var _ fssz.Marshaler = (*SSZUint64)(nil)
+var _ fssz.Unmarshaler = (*SSZUint64)(nil)
+
+// SSZUint64 is a uint64 type that satisfies the fast-ssz interface.
+type SSZUint64 uint64
+
+// MarshalSSZTo marshals the uint64 with the provided byte slice.
+func (s *SSZUint64) MarshalSSZTo(dst []byte) ([]byte, error) {
+	marshalled, err := s.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	return append(dst, marshalled...), nil
+}
+
+// MarshalSSZ marshals uin64 into a serialized object.
+func (s *SSZUint64) MarshalSSZ() ([]byte, error) {
+	marshalled := fssz.MarshalUint64([]byte{}, uint64(*s))
+	return marshalled, nil
+}
+
+// UnmarshalSSZ deserializes the provided bytes buffer into the uint64 object.
+func (s *SSZUint64) UnmarshalSSZ(buf []byte) error {
+	if len(buf) != s.SizeSSZ() {
+		return fmt.Errorf("expected buffer of length %d received %d", s.SizeSSZ(), len(buf))
+	}
+	*s = SSZUint64(fssz.UnmarshallUint64(buf))
+	return nil
+}
+
+// HashTreeRoot returns calculated hash root.
+func (s SSZUint64) HashTreeRoot() ([32]byte, error) {
+	return fssz.HashWithDefaultHasher(s)
```

Comment:
lets keep this as it was before
```suggestion
	return htrutils.Uint64Root(uint64(*s)), nil
```

Reply:
I copied the specific `htrutils` and `bytesutil` implementation to this function instead of copying the full package over. Originally try to make the implementation of `SSZUint64` looks like other ssz `uint64` (e.g. `Slot` and `Epoch`), and the copied test passed with new implementation. Any concern why we can't make the change?

Reply:
Mostly just to leave the implementation the same as the one we have now. Also another point is that it can create unecessary instances of `sha256` hashers. If we are constantly hashing this object, a new 32 byte buffer and sha256 hasher instance are created. Which is kind of unnecessary for what is a simple uint64. The other implementations might not be efficient if they are using the same default logic here too.